### DISCUSSION TITLE
First-time user experience

### DISCUSSION THREAD

**Discussion by ghost** [marlonrichert/zsh-autocomplete#376] [General]
Tried this out again after about two years (maybe?) to see if I could integrate it into my workflow and want to provide my user experience this time around as last time I didn't have the time to provide feedback.

Installation:

- Unclear with package managers or package names used as not covered in the README
- I installed one of the AUR packages which has a higher rating and points to this repo
- Manually added configuration (I'd expect this to be automated to some degree, esp. for evaluation)
- Manually sourced from `.zshrc` alongside Powerlevel10k and disabled completions as instructed

Observations:

-  Plugin added a noticeable delay of about 2-3 seconds after shell starts to <kbd>Ctlr+L</kbd>. So I reviewed my config to make sure completions were disabled. Same experience on shell startup.
- I'm not able to see the experience shown in the GIF where typeahead is shown. I'm currently installing a different plugin from zsh-users for that.
- Semicolons are appearing at the end of commands I type from history, which affects program output
- Pressing tab to automatically accepts the first item in the list. I expect it to cycle through the list.
- It feels like my cursor is constantly being hijacked and moves around expectedly while typing
- The perceived latency of my connectedness with my command line is diminished
- I was unable to properly configure the `'..##'` setting to ignore autocompletion of repeating full-stops

How I tested:

- Created a PRoot environment on my Android device using Termux
- Installed Arch Linux using `proot-distros`
- Installed Powerlevel10k
- Installed this Plugin

Feedback:

- Please document the names of the packages used by the package managers to ease installation.
- Find a low-friction method for allowing users to try before they buy.

**Comment by marlonrichert** [comment]
Hi! Thanks for your feedback. 🙂

> * Unclear with package managers or package names used as not covered in the README
> * Please document the names of the packages used by the package managers to ease installation.

I don't maintain any packages for my plugins. The recommended ways to install are [documented in the Readme](https://github.com/marlonrichert/zsh-autocomplete#installing--updating).

> Manually added configuration (I'd expect this to be automated to some degree, esp. for evaluation)

You shouldn't have to add any configuration, unless you want to change the defaults. Everything works out of the box.

> I'm not able to see the experience shown in the GIF where typeahead is shown.

Thanks for reporting. I should update that Gif; it's quite old. Added to my todo list: #377

> Semicolons are appearing at the end of commands I type from history, which affects program output

They should disappear automatically when you type, move the cursor or press enter; see #370. If they don't, then please open a separate bug report for that.

> Pressing tab to automatically accepts the first item in the list. I expect it to cycle through the list.

That's configurable: https://github.com/marlonrichert/zsh-autocomplete/blob/ccf7d5947635b89c5dab77d60538a950ff001baf/.zshrc#L41

> * Plugin added a noticeable delay of about 2-3 seconds after shell starts to `Ctlr`+`L`. So I reviewed my config to make sure completions were disabled. Same experience on shell startup.
> * It feels like my cursor is constantly being hijacked and moves around expectedly while typing
> * The perceived latency of my connectedness with my command line is diminished
> * I was unable to properly configure the '..##' setting to ignore autocompletion of repeating full-stops

I'm unable to reproduce any of this. Please open separate bug reports for these.

> Find a low-friction method for allowing users to try before they buy.

Not sure what you mean by that; all you need to do is `git clone` the repo and `source` the `.plugin.zsh` file, as is usual for any Zsh plugin. Can you be more specific?

**Comment by ghost** [comment]
Thanks for considering my feedback @marlonrichert. I'm still very much interested in finding a way to incorporate the functionality you've cooked into this software into my development workflow. All things being equal, what you provide is a great developer affordance and so I hope by improving the docs a little you'll start getting more feedback to find ways to continue improving this tool.

Regarding the GIF, instead of removing it perhaps you could simply point users as complimentary tooling (even if it carries a very similar name—also a point of confusion). Re: The perceived latency I'm working on moving my development inside a chroot jail on my Android phone so what I was experiencing was akin to what this plugin would feel like working over SSH (as I'm working over SSH on my LAN to build out my development environment using PRoot on Android).

Re: The package managers. Rather then eschewing them users could have a better experience during set-up with increased more detailed documentation. Stating you aren't maintaining them is very understandable, and I don't believe users would mind submitting tickets package manager tickets on your behalf as a way of saying thanks for the tooling. 😃
